packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -398,13 +477,51 @@ function WorklistDataGrid(props) {
     }
   };
 
-  const handleCellSelection = (cellRowIdColId, cellCoordinates) => {
-    selectCell(cellRowIdColId, cellCoordinates);
+  const handleCellSelection = (selectionDetails) => {
+    selectCell(selectionDetails);
   };
 
-  const handleRowSelection = (rowId, rowIndex, selectedCellCoordinates) => {
-    handleCellSelectionChange(null, null, selectedCellCoordinates);
-    selectRows(false, rowId, rowIndex);
+  const handleRowSelection = (selectionDetails) => {
+    handleCellSelectionChange(null, null, { row: selectionDetails.rowIndex, col: selectionDetails.columnIndex });
+    const selectionData = selectionDetails.selectedByKeyboard ? rowSelection.current.lastIndividuallySelectedRow : rowSelection?.current?.focussedRow;
+
+    if (!selectionDetails.multiSelect) {
+      if (hasSelectableRows) {
+        // regular click or space key
+        selectRows(false, selectionDetails.rowId, selectionDetails.rowIndex);
+      }
+    } else if (hasSelectableRows) {
+      // Shift key is pressed
+      if (selectionData?.anchorRow) {
+        // ACTION: select multiple rows based on anchor
+        selectMultipleRows(selectionDetails.rowIndex, selectionData);
+        selectionData.previousSelectionEndRow = selectionDetails.rowIndex;
+      } else {
+        // no anchor row
+        selectRows(false, selectionDetails.rowId, selectionDetails.rowIndex);
+      }
+    } else if (selectionDetails.selectedByKeyboard) {
+      // Shift + Space
+      rowSelection.current.isRowSelectionEnabledByGrid = !hasSelectableRows;
+      rowSelection.current.lastIndividuallySelectedRow = { anchorRow: selectionDetails.rowIndex };
+      selectMultipleRows(selectionDetails.rowIndex, rowSelection.current.lastIndividuallySelectedRow);
+      rowSelection.current.lastIndividuallySelectedRow.previousSelectionEndRow = selectionDetails.rowIndex;
```

Comment:
Why does the lastIndividuallySelectedRow need to be an object?  There seem to be quite a few layers to maintain this new logic.

Reply:
We updated the name of the variable to rangeUsingLastIndividuallySelectedRow since it contains the anchor for the range and the end point in the range.